inst/tests/mergelist.Rraw

```diff
@@ -0,0 +1,81 @@
+require(methods)
+
+if (exists("test.data.table", .GlobalEnv, inherits=FALSE)) {
+  if ((tt<-compiler::enableJIT(-1))>0)
+    cat("This is dev mode and JIT is enabled (level ", tt, ") so there will be a brief pause around the first test.\n", sep="")
+} else {
+  require(data.table)
+  test = data.table:::test
+}
+
+# cbindlist
+
+local({
+  l = list(
+    d1 = data.table(x=1:3, v1=1L),
+    d2 = data.table(y=3:1, v2=2L),
+    d3 = data.table(z=2:4, v3=3L)
+  )
+  ans = cbindlist(l)
+  expected = data.table(l$d1, l$d2, l$d3)
+  test(11.01, ans, expected)
+  test(11.02, intersect(vapply(ans, address, ""), unlist(lapply(l, vapply, address, ""))), character())
+  ans = cbindlist(l, copy=FALSE)
+  expected = setDT(c(l$d1, l$d2, l$d3))
+  test(11.03, ans, expected)
```

Comment:
Isn't this more about testing `setDT` than about testing `cbindlist`?

Reply:
It's a mix of both, right? I'm not sure there's an easy way to avoid that...